### ISSUE TITLE
Bug: lint detection issue, causing error

### ISSUE THREAD

**Issue by Harley-Torrisi** [eslint/eslint#16642]
### Environment

Node version: v18.2
npm version: 8.19
Local ESLint version: 8.29.0 
Local ESLint version(Next): 13.0.6
Operating System: MacOS

### What parser are you using?

Other

### What did you do?

<details>
<summary>
When writing a function that has a return object of square braces, will get a lint error if the function does not have any property in the overload. It will expect the end brace to be indented by 1. This looks like an error as I can see no rule configuration that would differentiate. 

For now I have suppressed, but thought I should report :) 
</summary>

```
<!-- Paste your configuration here -->
{
	"extends": "next/core-web-vitals",
	"rules": {
		"indent": [
			"error",
			"tab"
		],
		"linebreak-style": [
			"error",
			"unix"
		],
		"unicode-bom": [
			"error",
			"never"
		],
		"quotes": [
			"error",
			"single"
		],
		"array-bracket-spacing": [
			"error",
			"never"
		],
		"brace-style": [
			"error",
			"allman",
			{
				"allowSingleLine": true
			}
		],
		"semi": [
			"error",
			"never"
		]
	}
}

</details>

```js
export function TestIncorrect(): [
	x: 10
] // <- error here
{
	return [10]
}

export function TestCorrect(value: string): [
	x: 10
]
{
	return [10]
}

```

### What did you expect to happen?

No error

### What actually happened?

Expected indentation of 1 tab but found 0.

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by yeonjuan** [comment]
Hi @Harley-Torrisi 
Your code seems typescript code. You should use `@typescript-eslint/indent`.

```json
"rules": {
		"indent": "off",
	        "@typescript-eslint/indent": ["error", "tab"],
 }
```

If you don't install the `@typescript-eslint` before, you should install and configure it first. https://typescript-eslint.io/getting-started

**Comment by Harley-Torrisi** [comment]
Cheers @yeonjuan 
Sorry for the mistaken bug report. Having defined TS when using the NextJS project builder, I assumed I got everything kitted as needed.
